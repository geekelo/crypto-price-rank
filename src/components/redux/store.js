import { configureStore } from '@reduxjs/toolkit';
import themeSlice from './theme/themeSlice';
import quotesSlice from './quotes/quotesSlice';
import cryptosSlice from './crypto/cryptosSlice';

const store = configureStore({
  reducer: {
    theme: themeSlice,
    quotes: quotesSlice,
    cryptos: cryptosSlice,
  },
});

export default store;
