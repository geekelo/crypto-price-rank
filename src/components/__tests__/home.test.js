import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter if needed
import store from '../redux/store';
import Home from '../home';

test('Home component matches snapshot', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    </Provider>,
  );

  // Compare the rendered component to a stored snapshot
  expect(asFragment()).toMatchSnapshot();
});
