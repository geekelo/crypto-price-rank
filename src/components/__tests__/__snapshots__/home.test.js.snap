// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <div
      class="landingarea lightlandingarea"
    >
      <div
        class="lightlandingbg"
      >
        <div>
          <form
            class="searchContain lightsearchContain"
          >
            <div
              class="input lightinput"
            >
              <input
                class="formField lightformField"
                placeholder="Search"
                type="search"
                value=""
              />
            </div>
            <div
              class="input lightinput"
            >
              <select
                class="formField lightformField"
              >
                <option
                  value="null"
                >
                  - Select Ranking -
                </option>
                <option
                  value="priceUsd"
                >
                  Rank by Price
                </option>
                <option
                  value="supply"
                >
                  Rank by Supply
                </option>
              </select>
            </div>
          </form>
        </div>
      </div>
    </div>
    <p
      class="allstats lightallstats"
    >
       CRYPTO RANKING 
    </p>
    <main>
      <ul
        class="listContain lightlistContain"
      />
    </main>
  </section>
</DocumentFragment>
`;
